File: python/src/problems1_100/remove_element.py
from abc import ABC, abstractmethod
from typing import override


class Problem(ABC):
    @abstractmethod
    def remove_element(self, nums: list[int], val: int) -> int:
        pass


class Default(Problem):
    @override
    def remove_element(self, nums: list[int], val: int) -> int:
        start = 0
        end = 0
        while start < len(nums):
            if nums[start] != val:
                nums[end] = nums[start]
                end += 1
            start += 1
        return end


SOLUTIONS: list[Problem] = [Default()]

File: typescript/src/problems1_100/remove-element.ts
class Problem {
  removeElement(nums: number[], val: number): number {
    throw new Error(`Not implemented (${nums}, ${val}).`);
  }
}

class Default extends Problem {
  removeElement(nums: number[], val: number): number {
    let start: number = 0;
    let end: number = 0;
    while (start < nums.length) {
      if (nums[start] !== val) {
        nums[end++] = nums[start];
      }
      start++;
    }
    return end;
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;
